docs/content/testing/writing-a-new-test.md

```diff
@@ -0,0 +1,314 @@
+        Assert.AreEqual(4, MyClass.AddTwoNumbers(3, 1));
+        Assert.AreEqual(7, MyClass.AddTwoNumbers(2, 5));
+    }
+}
+```
+
+For the instance functionality, you can add tests similarly:
+
+```cs
+[TestFixture]
+public class MyClassInstanceTests
+{
+    private MyClass myClassInstance;
+
+    [SetUp]
+    public void SetUp()
+    {
+        myClassInstance = new MyClass();
+    }
+
+    [TearDown]
+    public void TearDown()
+    {
+        myClassInstance.Dispose();
+    }
+
+    [Test]
+    public void InstanceValue_is_0_by_default()
+    {
+        Assert.AreEqual(0, myClassInstance.InstanceValue);
+    }
+
+    [Test]
+    public void DoInstanceOperation_increases_InstanceValue_by_1()
+    {
+        myClassInstance.DoInstanceOperation();
+        Assert.AreEqual(1, myClassInstance.InstanceValue);
+
+        myClassInstance.DoInstanceOperation();
+        Assert.AreEqual(2, myClassInstance.InstanceValue);
+    }
+
+    [Test]
+    public void DoSomethingElse_increses_InstanceValue_by_5()
+    {
+        myClassInstance.DoSomethingElse();
+        Assert.AreEqual(5, myClassInstance.InstanceValue);
+
+        myClassInstance.DoSomethingElse();
+        Assert.AreEqual(10, myClassInstance.InstanceValue);
+    }
+}
+```
+
+### Internal variables and methods
+
+The tests above will not compile. This is because the test class is in a different assembly to the implementation class, and the tests are looking at internal fields.
+
+To solve this; make the internals of the the `MyClass.cs` file visible to the test assembly in the way shown below.
+
+```
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Improbable.Gdk.Core.EditModeTests")]
+```
+
+### Running the new tests
+
+In the Test Runner window of Unity, find your test under: **Improbable.Gdk.Core.EditModeTests.dll**. (This matches the folder name
+ `Gdk/Core/Tests/EditMode`.)
+
+You can follow the namespace and fixture name in the hierarchy:
+
+- **Improbable** > ** Gdk**  > ** Core**  > ** EditModeTests** > ** Utility**  > ** MyClassStaticTests** 
+- ** Improbable**  > ** Gdk**  > ** Core**  > ** EditModeTests**  > ** Utility**  > ** MyClassInstanceTests** 
+
+Double-click the test fixture name to run all tests within that fixture.
+
+## How to create a new test folder and assembly
+
+To add tests for the `/workers/Unity/Assets/Gdk/Legacy` folder within the GDK Unity project's assets:
+
+1. Create the directory: `/workers/Unity/Assets/Gdk/Legacy/Tests/EditMode`
+
+2. In the Unity Editor’s Project window, right-click the directory in the Project and select **Create** > **Assembly Definition**.
+
+3. Name the new **Assembly Definition** file you have just created to match: `Improbable.Gdk.Legacy.EditModeTests`.
+
+4. Select this file, and change the **name** property in the Unity Inspector to match the filename:`Improbable.Gdk.Legacy.EditModeTests`.
+
+5. Click  **Apply** .
+
+6. Check the **Test Assemblies** checkbox.
+
+7. If you're writing EditMode tests:
```

Comment:
These 0. points are not rendering as separate lines,

Reply:
Fixed